### ISSUE TITLE
[MU4 Issue] Staff text and Dynamics are not duplicated to linked staff

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#11877]
**To Reproduce**
1. Create score
2. Add linked staff
3. Add notes 
4. Add staff text dynamics to one staff - **it's not duplicated to other staff**

https://user-images.githubusercontent.com/90187801/171845982-b44dfe1e-0925-454d-89d1-31c282bda397.mp4

**Platform information**
macOS, Windows, Linux

**Additional context**
The same in **MU3**

**Comment by Tantacrul** [comment]
@MarcSabatella - can you give us any insight into this? I can't quite imagine if this was decided upon or whether it's an issue. Obviously, it's weird that diminuendo persists but not dynamics.

**Comment by MarcSabatella** [comment]
Interesting!  Offhand, I don't recall this ever having been decided consciously, but the original development of linked staves happened before I got very involved.

I don't know that in general the default approach has been, duplicate everything and let people mark them invisible on one staff or the other.  There have been a few notable exceptions, like fingering, which originally was *not* duplicated but enough requests came in for it that this was made an explicit option in the properties for tab staves.  And that's probably a reasonable approach for other element types too, so if you know you never want dynamics or hairpins, you can just set that once and be done.

What I can see is that almost *no* text is duplicated on the tab staff in MU3 - not dynamics, not staff text, not chord symbols, not lyrics.  Instrument changes are, and so is sticking - and notably, these are two text types that were not introduced until after the original development of the linked staff mechanism.

I did a quick search of the forum and found a few discussions of the topic.  I wouldn't say there seems to be a very strong consensus one way or the other, but overall, people are mostly satisfied with text not being linked.  See for example https://musescore.org/en/node/287231.  Also this much older discussion ("Miwarre" was the original designer/developer of the tab feature) - https://musescore.org/en/node/32431

Bottom line, I think I'd recommend not linking hairpins to make things consistent without rocking the boat too much.

**Comment by DmitryArefiev** [comment]
@MarcSabatella Oh I see now.. If we use linked staff only as a Tab staff then we should exclude (for duplicating) **hairpins** and also **some other lines** . 

https://user-images.githubusercontent.com/90187801/171916870-a75bbf5f-cedd-4578-bc22-0d7abc74d6fe.mp4


If user wants to add some specific lines he can add it independently.

@Tantacrul If agree I'll revert the bug

**Comment by Tantacrul** [comment]
I have no issues with this in principal. The only concern I have is that removing the linkage of hairpins might cause issues for Guitar Pro import in MS4.0.

@alexpavlov96 & @boryskuzmenko what do you think?

If we want to be cautious, I'd suggest bumping this bug until 4.x.

**Comment by MarcSabatella** [comment]
That makes sense to me, yes.  Although I wonder about the "barre line" - the one with the VII as the default text.  Does that make sense on tab?  I don't know.

**Comment by boryskuzmenko** [comment]
This is not going to affect GP import, BUT.  
As I understand right only case where linked staff can be useful for now is linking standard+TAB. So I see the same solution exactly like we did for GP import. First, create a table with expected states for all groups of elements in particular staves combination (standard-standard, standard-TAB Simple, Standard-TAB Common) e.g. https://docs.google.com/spreadsheets/d/1eZB5gw7nqKFcpqoeI-NpcF493WBf4ldHd8m_OGMsi48/edit#gid=0. Then to implement this as a style preferences which applied  depends on particular staves combination. 
<img width="953" alt="Styles_guitar-tablature-v 4a" src="https://user-images.githubusercontent.com/101250347/172011501-9a6f5fc6-c371-4900-8028-f05e5b0ffcbb.png">

This way looks complicated but covers everything we want and gives to user an ability to customize this rules for all linked staves on score.
Even if we’re not going to implement as preferences I beg to create a table with expected behaviors. In this case we at least will be on the same page.
So, my opinion is better to move it to MU4.x and decided how to be with that then.

**Comment by Tantacrul** [comment]
I would agree that we make decisions on this in 4.X. It is a tad dangerous to be fiddling with stuff already established in MS3 at this late stage.

I've elevated it to a P1 because I think dealing with this consistently is going to be very important to getting our Guitar experience right.

**Comment by adamspagge** [comment]
To support @abariska point about needing different behaviors for "Standard-Standard" versus "Standard-TAB" linked staves, I'd like to highlight a use case that requires text linkage: historical vocal transcriptions. I would use Standard-Standard linked staves to maintain a master score with the original C-clef notation (for archival comparison) simultaneously linked to a modern G-clef performance part (for singers).

(As a side note on current functionality: At first glance, creating Standard-Standard linked staves with different clefs seems impossible. However, in MU4.7 at least, you can achieve this workaround by changing the clef for both staves and then manually deleting only one of them to create the C-clef/G-clef combination.)

In this specific scenario, being able to link lyrics—which @MarcSabatella noted currently do not link—along with dynamics and staff text is essential, unlike in a Standard-TAB setup where they might just add clutter. Implementing customizable linking rules or expected states based on the specific staff combination would perfectly accommodate this workflow.

On a related note, since the handbook doesn't currently provide clear insight into expected behaviour, it would be very helpful to update the documentation. Explicitly stating that lyrics and other text elements are not linked by design would clear up a lot of confusion for users trying to figure out if they are encountering a bug or an intended feature. (And maybe also adding the workaround for different clefs).
